components/uniset/src/enum_props.rs

```diff
@@ -129,337 +144,387 @@ pub enum CanonicalCombiningClass {
     CCC91 = 91,
 }
 
+/// Decomposition Type constants.
+/// These are listed as Compatibility Formatting Tags in
+/// section 5.7.3 "Character Decomposition Mapping" in Unicode Standard Annex #44,
+/// "Unicode Character Database".
+/// See UDecompositionType in ICU4C.
 #[derive(Clone, PartialEq, Debug)]
 pub enum DecompositionType {
-    Can,
-    Com,
-    Enc,
-    Fin,
-    Font,
-    Fra,
-    Init,
-    Iso,
-    Med,
-    Nar,
-    Nb,
-    None,
-    Sml,
-    Sqr,
-    Sub,
-    Sup,
-    Vert,
-    Wide,
+    Can = 1,
+    Com = 2,
+    Enc = 3,
+    Fin = 4,
+    Font = 5,
+    Fra = 6,
+    Init = 7,
+    Iso = 8,
+    Med = 9,
+    Nar = 10,
+    Nb = 11,
+    None = 0,
+    Sml = 12,
+    Sqr = 13,
+    Sub = 14,
+    Sup = 15,
+    Vert = 16,
+    Wide = 17,
 }
 
+/// A property for determining the choice of wide versus narrow glyphs in East Asian contexts.
+/// Property values are described in Unicode Standard Annex #11, "East Asian Width".
+/// See UEastAsianWidth in ICU4C.
 #[derive(Clone, PartialEq, Debug)]
 pub enum EastAsianWidth {
-    Ambiguous,
-    Fullwidth,
-    Halfwidth,
-    Neutral,
-    Narrow,
-    Wide,
+    Ambiguous = 1,
+    Fullwidth = 3,
+    Halfwidth = 2,
+    Neutral = 0,
+    Narrow = 4,
+    Wide = 5,
 }
 
+// TODO: Ask how to create a stable numbering for enums not 
+// represented in ICU4C (ICU4C has ~ 30 enums, PPUCD has ~ 38
+// enums). Hunch: the "extra" PPUCD properties are the "rollup"
+// properties.
+// TODO: ^ If hunch is true, find docs for the precise terminology.
+// TODO: make proposal on how to handle. Have a separate enum for
+// rollups? If so, what about Letter vs. CasedLetter vs. LowercaseLetter?
+// TODO: Why does ICU4C have `Other` and `Unassigned` use the same 
+// enum value (discriminant) ?
+//
+/// Enumerated Unicode general category types.
+/// See https://www.unicode.org/reports/tr44/ .
+/// See UCharCategory in ICU4C.
 #[derive(Clone, PartialEq, Debug)]
 pub enum GeneralCategory {
-    Other,
-    Cntrl,
-    Format,
-    Unassigned,
-    PrivateUse,
-    Surrogate,
-    Letter,
-    CasedLetter,
-    LowercaseLetter,
-    ModifierLetter,
-    OtherLetter,
-    TitlecaseLetter,
-    UppercaseLetter,
-    CombiningMark,
-    SpacingMark,
-    EnclosingMark,
-    NonspacingMark,
-    Number,
-    Digit,
-    LetterNumber,
-    OtherNumber,
-    Punct,
-    ConnectorPunctuation,
-    DashPunctuation,
-    ClosePunctuation,
-    FinalPunctuation,
-    InitialPunctuation,
-    OtherPunctuation,
-    OpenPunctuation,
-    Symbol,
-    CurrencySymbol,
-    ModifierSymbol,
-    MathSymbol,
-    OtherSymbol,
-    Separator,
-    LineSeparator,
-    ParagraphSeparator,
-    SpaceSeparator,
+    Other = 0,
+    Cntrl = 15,
+    Format = 16,
+    Unassigned = 0,
+    PrivateUse = 17,
+    Surrogate = 18,
+    // Letter = ,
```

Comment:
You can't put the multi-category set values into the enum with the single General_Category values.

What you could do is what ICU does, and define "mask" values for each General_Category with one bit each, and multi-bit constants for things like "Letter". ICU has C API for this, using numeric values typed as uint32_t.

The friendly equivalent in modern Java would be an EnumSet. You could just use things like `EnumSet.of(OtherLetter)` and define an EnumSet like LetterSet and use it like `LetterSet.contains(getGeneralCategory(c))`.

Does Rust have something like EnumSet?
https://docs.oracle.com/javase/8/docs/api/java/util/EnumSet.html

(Internally, it efficiently uses a bit set, just like the ICU C API does explicitly but in poor man's style.)

Reply:
Done. The multi-category set values were removed, and there will be work later to properly support those multi-category set values via an equivalent to EnumSet.